### PR TITLE
Add unwrap method to recursive convert to plain old python objects

### LINKED ISSUE #43: Function to convert everything to plain old python objects
As more projects incorporate PEP518, many will be aiming to keep compatibility with their existing config files as well (be they ini, yaml, json, etc.). In order to help abstract over config files of different types, it would be useful to be able to convert tomlkit objects into POPOs. One of TOML's strengths for python is that all of its constructs can deserialise to POPOs: could this be made available for tomlkit users?

### PR DESCRIPTION
**Pull request by syntapy**
Adds methods to convert TOMLKit objects to plain python methods

Fixes #43

### CODE REVIEW

File: tomlkit/container.py [hunk 1]
```diff
@@ -46,6 +46,25 @@ def __init__(self, parsed: bool = False) -> None:
     def body(self) -> List[Tuple[Optional[Key], Item]]:
         return self._body
 
+    def unwrap(self, recursive: bool = True) -> str:
+        unwrapped = {}
+        for k, v in self._body:
```
Comment (frostming) [inline tomlkit/container.py:51]: We don't need to iterate over the `_body`, `for k, v in self.items()` can retrieve all key and value pairs, without any format control elements.

File: tomlkit/items.py [hunk 2]
```diff
@@ -468,6 +472,8 @@ def __reduce__(self):
     def __reduce_ex__(self, protocol):
         return self.__class__, self._getstate(protocol)
 
+    def 
+
```
Comment (frostming) [inline tomlkit/items.py:476]: Need to be cleaned up.

File: tomlkit/items.py [hunk 3]
```diff
@@ -957,6 +984,15 @@ def __init__(self, value: list, trivia: Trivia, multiline: bool = False) -> None
         self._multiline = multiline
         self._reindex()
 
+    def unwrap(self, recursive: bool = True) -> str:
```
Comment (frostming) [inline tomlkit/items.py:987]: Are there any cases that we don't want recursive unwrap? `list(array)` should do the trick.

**Review by frostming — CHANGES_REQUESTED** [review]

**Comment by syntapy** [comment]
Ok I think I just need to add tests for
- [ ] `container.Container`
- [ ] `container.OutOfOrderTableProxy`
- [ ] `items.Item`
- [ ] `items.Integer`
- [ ] `items.Float`
- [ ] `items.Bool`
- [ ] `items.DateTime`
- [ ] `items.Date`
- [ ] `items.Time`
- [ ] `items.Array`
- [ ] `items.AbstractTable`
- [ ] `items.String`
- [ ] `items.AoT`
- [ ] `items.Null`

File: tomlkit/items.py (hunk 3 above)
Comment (syntapy) [reply tomlkit/items.py:1079]: So you mean just returning `list(self)`?

> Are there any cases that we don't want recursive unwrap?

Is that only applicable to the `items.Array` object or do you want me to make all of of the `unwrap` methods recursive?

Comment (syntapy) [reply tomlkit/items.py:1079]: Ok, I think I see something of what you're saying. 

So if a create an `Array` using `array = items.Array([item(2), item(2.5), item(False)], trivia])`, the third one (boolean) gets turned into a ppo but not the others

This makes the recursive `unwrap` fail on that

But if I do `return list(self)` then the `Integer` and `Float` items in indices 0 and 1 don't come out as ppo

Comment (frostming) [reply tomlkit/items.py:1079]: I mean `recursive=False` is equal to `list(array)`, so we don't need this optional flag and make all `unwrap` recursive.

File: tomlkit/container.py [hunk 4]
```diff
@@ -46,6 +48,25 @@ def __init__(self, parsed: bool = False) -> None:
     def body(self) -> List[Tuple[Optional[Key], Item]]:
         return self._body
 
+    def unwrap(self) -> str:
+        unwrapped = {}
+        for k, v in self.items():
+            if k is None:
+                continue
+
+            if not type(k) == str:
```
Comment (frostming) [inline tomlkit/container.py:57]: `type(k) == str` is not recommended against `isinstance`. For this check, we can replace to `isinstance(Key)`

File: tomlkit/container.py [hunk 5]
```diff
@@ -46,6 +48,25 @@ def __init__(self, parsed: bool = False) -> None:
...
+                k = k.key
+
+            if not type(v) == str:
+                v = v.unwrap()
```
Comment (frostming) [inline tomlkit/container.py:61]: Might be better to call `unwrap()` and catch `AttributionError`.

**Review by frostming — CHANGES_REQUESTED** [review]

File: tomlkit/container.py (hunk 4 above)
Comment (syntapy) [reply tomlkit/container.py:57]: ok I have done that

now its that because of a possibility that `k` and `v` are tomlkit types  that inherit from `str`?

Comment (syntapy) [reply tomlkit/container.py]: I think I need to do some similar stuff with the tests that use the `type(..)` function. I will let you know when I get that done in the next day or so

File: tomlkit/container.py [hunk 6]
```diff
@@ -46,42 +46,33 @@ def __init__(self, parsed: bool = False) -> None:
...
-                continue
+    def _unwrap_inner(self, k, v, d):
+        if k is None:
+            return
 
-            k = k.key
+        k = k.key
+        v = v.value
+
+        if isinstance(v, Container):
```
Comment (frostming) [inline tomlkit/container.py]: It is safe to use `isinstance(v, Item)` here since the child of a container can't be a container.

Comment (syntapy) [reply tomlkit/container.py]: I tried that, not sure I did it correctly

File: tomlkit/items.py [hunk 7]
```diff
@@ -1675,6 +1728,15 @@ def __init__(
         for table in body:
             self.append(table)
 
+    def unwrap(self) -> str:
+        unwrapped = []
+        for t in self._body:
+            if is_tomlkit(t):
```
Comment (frostming) [inline tomlkit/items.py:1734]: Also, a child of an array must be an item. `isinstance(t, Item)` is enough for this case. But this also works

Comment (syntapy) [reply tomlkit/items.py]: Done

**Review by frostming — APPROVED** [review]

**Comment by syntapy** [comment]
@frostming Is it OK if I make a PR to bump the version? If so which version? 0.11.0 or 0.10.2?

I am trying to use the version of tomlkit with these changes in a [PR](https://github.com/iterative/dvc/pull/7468)

**Pull request merged**